openpype/hosts/max/plugins/publish/save_scene.py

```diff
@@ -1,21 +1,23 @@
 import pyblish.api
-import os
+from openpype.pipeline import registered_host
 
 
 class SaveCurrentScene(pyblish.api.ContextPlugin):
-    """Save current scene
-
-    """
+    """Save current scene"""
 
     label = "Save current file"
     order = pyblish.api.ExtractorOrder - 0.49
     hosts = ["max"]
     families = ["maxrender", "workfile"]
 
     def process(self, context):
-        from pymxs import runtime as rt
-        folder = rt.maxFilePath
-        file = rt.maxFileName
-        current = os.path.join(folder, file)
-        assert context.data["currentFile"] == current
-        rt.saveMaxFile(current)
+        host = registered_host()
```

Comment:
I spend some time reading this again and maybe we dont need to access the workfilehost to do all these codes. Consider the code below :) :
```
import os
import pyblish.api

class SaveCurrentScene(pyblish.api.ContextPlugin):
    """Save current scene"""

    label = "Save current file"
    order = pyblish.api.ExtractorOrder - 0.49
    hosts = ["max"]
    families = ["maxrender", "workfile"]

    def process(self, context):
        from pymxs import runtime as rt
        folder = rt.maxFilePath
        file = rt.maxFileName
        current = os.path.join(folder, file)
        assert context.data["currentFile"] == current
        if rt.getSaveRequired():
            rt.saveMaxFile(current)
        else:
            self.log.debug("No unsaved changes, skipping file save..")

```

Reply:
That's true although those function exist on the host to be easily reused, no?  My assumption was that they exist for new validators to be easily implemented without extensive knowledge of the host. You can simply use another DCC/Host as an example without needing to learn 3dsmax's API to redefine rt.getSaveRequired() (and any others) in each validator that would require it.  

I can't say I have a huge preference here as I know the DCC and it's API implementation but why then do the function exist on the host, if not to be used?

Reply:
I see your point but if it can be re-used, which means the context plugins can be also used in other hosts (maya, max, houdini) with `workfile_has_unsaved_changes()`. Any thoughts? @BigRoy @antirotor

Reply:
https://github.com/ynput/OpenPype/blob/77f59eb9ef316e9d628f7ee4df0acf5e40d419a9/openpype/host/interfaces.py#L203
https://github.com/ynput/OpenPype/blob/77f59eb9ef316e9d628f7ee4df0acf5e40d419a9/openpype/hosts/houdini/api/pipeline.py#L88-L89
https://github.com/ynput/OpenPype/blob/77f59eb9ef316e9d628f7ee4df0acf5e40d419a9/openpype/hosts/houdini/plugins/publish/save_scene.py#L13-L26

Base class and Houdini implementation for ref